main.nf

```diff
@@ -0,0 +1,708 @@
+                                          ,--./,-.
+          ___     __   __   __   ___     /,-._.--~\'
+    |\\ | |__  __ /  ` /  \\ |__) |__         }  {
+    | \\| |       \\__, \\__/ |  \\ |___     \\`-._,-`-,
+                                          `._,._,\'
+
+nf-core/rnafusion v${workflow.manifest.version}"
+======================================================="""
+def summary = [:]
+summary['Pipeline Name']  = 'nf-core/rnafusion'
+summary['Pipeline Version'] = workflow.manifest.version
+summary['Run Name']     = custom_runName ?: workflow.runName
+summary['Reads']        = params.reads
+summary['Fasta Ref']    = params.fasta
+summary['Data Type']    = params.singleEnd ? 'Single-End' : 'Paired-End'
+summary['Max Memory']   = params.max_memory
+summary['Max CPUs']     = params.max_cpus
+summary['Max Time']     = params.max_time
+summary['Output dir']   = params.outdir
+summary['Working dir']  = workflow.workDir
+summary['Container Engine'] = workflow.containerEngine
+if(workflow.containerEngine) summary['Container'] = workflow.container
+summary['Current home']   = "$HOME"
+summary['Current user']   = "$USER"
+summary['Current path']   = "$PWD"
+summary['Working dir']    = workflow.workDir
+summary['Output dir']     = params.outdir
+summary['Script dir']     = workflow.projectDir
+summary['Config Profile'] = workflow.profile
+if(workflow.profile == 'awsbatch'){
+   summary['AWS Region'] = params.awsregion
+   summary['AWS Queue'] = params.awsqueue
+}
+if(params.email) summary['E-mail Address'] = params.email
+log.info summary.collect { k,v -> "${k.padRight(15)}: $v" }.join("\n")
+log.info "========================================="
+
+
+def create_workflow_summary(summary) {
+
+    def yaml_file = workDir.resolve('workflow_summary_mqc.yaml')
+    yaml_file.text  = """
+    id: 'nf-core-rnafusion-summary'
+    description: " - this information is collected when the pipeline is started."
+    section_name: 'nf-core/rnafusion Workflow Summary'
+    section_href: 'https://github.com/nf-core/rnafusion'
+    plot_type: 'html'
+    data: |
+        <dl class=\"dl-horizontal\">
+${summary.collect { k,v -> "            <dt>$k</dt><dd><samp>${v ?: '<span style=\"color:#999999;\">N/A</a>'}</samp></dd>" }.join("\n")}
+        </dl>
+    """.stripIndent()
+
+   return yaml_file
+}
+
+/*************************************************************
+ * Fusion pipeline
+ ************************************************************/
+
+/*
+ * STAR-Fusion
+ */
+process star_fusion {
+    tag "$name"
+    publishDir "${params.outdir}/tools/StarFusion", mode: 'copy'
+
+    when:
+    params.star_fusion || (params.star_fusion && params.test)
+
+    input:
+    set val(name), file(reads) from read_files_star_fusion
+    file reference from star_fusion_ref.collect()
+
+    output:
+    file '*fusion_predictions.tsv' into star_fusion_fusions
+    file '*' into star_fusion_output
+
+    script:
+    if (params.singleEnd) {
```

Comment:
Have you tried something like:
```
option = params.singleEnd ? : "--right_fq ${reads[1]}"
"""
STAR-Fusion \\
  --genome_lib_dir ${reference} \\
  --left_fq ${reads[0]} \\
  ${option} \\
  --CPU  ${task.cpus} \\
  --examine_coding_effect \\
  --output_dir .
"""
```
To simplify the code?

Reply:
Nice! I like it. Implemented